### PR TITLE
Migrate workflows to GH Actions

### PR DESCRIPTION
**Pull request by nihal-deriv**
Migrate the workflows from cicleCI  to github actions

### CODE REVIEW

File: .github/workflows/build_workflow.yml [hunk 1]
```diff
@@ -1,18 +1,44 @@
...
-      - run: echo "💡 The ${{ github.repository }} repository has been cloned to the runner."
-      - run: echo "🖥️ The workflow is now ready to test your code on the runner."
-      - name: List files in the repository
-        run: |
-          ls ${{ github.workspace }}
-      - run: echo "🍏 This job's status is ${{ job.status }}."
+      -
+        name: Checkout code
+        uses: actions/checkout@v4
+      -
```
Comment (lim-deriv) [inline .github/workflows/build_workflow.yml:19]: Could we remove the empty line to make it more tidy?

Comment (nihal-deriv) [reply .github/workflows/build_workflow.yml:19]: I think this format is neater and also improves readability

File: .github/workflows/build_workflow.yml [hunk 2]
```diff
@@ -1,18 +1,44 @@
...
+        name: Build Docker Image
+        uses: docker/build-push-action@0565240e2d4ab88bba5387d719585280857ece09
```
Comment (lim-deriv) [inline .github/workflows/build_workflow.yml:21]: May I know why you specifically choose this version of docker/build-push-action ? 
https://github.com/docker/build-push-action over here it shows the latest version is @v5 .

Comment (nihal-deriv) [reply .github/workflows/build_workflow.yml:21]: security team suggested we use full length SHA commits when using third party actions. This SHA commit corresponds to the latest commit under the v5 tag.

File: .github/workflows/build_workflow.yml [hunk 3]
```diff
@@ -1,18 +1,44 @@
...
+        with:
+          context: 'dzil/'
+          tags: deriv/dzil:latest
+          push: false   
+      -
+        name: docker image info
```
Comment (lim-deriv) [inline .github/workflows/build_workflow.yml:41]: do we need this step?

File: .github/workflows/hadolint.yml [hunk 4]
```diff
@@ -0,0 +1,20 @@
+on:
+  workflow_call:
+jobs:
+  hadolint:
+    runs-on: ubuntu-latest
+    steps:
+      - 
+        name: Checkout Code
+        uses: actions/checkout@v3
```
Comment (lim-deriv) [inline .github/workflows/hadolint.yml:9]: Is there any reason why this actions/checkout is using v3 version?

File: .github/workflows/hadolint.yml [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+      -
+        name: Install hadolint
+        run: | 
+          sudo wget -O /usr/bin/hadolint https://github.com/hadolint/hadolint/releases/download/v2.7.0/hadolint-Linux-x86_64
```
Comment (lim-deriv) [inline .github/workflows/hadolint.yml:13]: Just curious why we are using v2.7.0, I saw the latest hadolint version is v2.12.0 over here https://github.com/hadolint/hadolint/releases?utm_source=thenewstack&utm_medium=website&utm_content=inline-mention&utm_campaign=platform .

File: .github/workflows/build_workflow.yml (hunk 3 above)
Comment (nihal-deriv) [reply .github/workflows/build_workflow.yml:41]: its a debug step. Sorry i forgot to remove it. Will update it

File: .github/workflows/hadolint.yml (hunk 4 above)
Comment (nihal-deriv) [reply .github/workflows/hadolint.yml:9]: will upgrade it to v4

File: .github/workflows/hadolint.yml (hunk 5 above)
Comment (nihal-deriv) [reply .github/workflows/hadolint.yml:13]: oh thanks for pointing it out. Will switch to the latest version

**Comment by mukesh-deriv** [comment]
Can we remove the circleCI config as well part of this PR. Its hard to get binary-com PRs to merge :(

**Pull request merged**